doc/sphinx/source/esmvalcore/preprocessor.inc

```diff
@@ -1,36 +1,218 @@
+The ESMValTool preprocessor can be used to perform a broad range of operations
+on the input data before diagnostics or metrics are applied. The
+preprocessor performs these operations in a centralized, documented and
+efficient way, thus reducing the data processing load on the diagnostics side.
+
+Each of the preprocessor operations is written in a dedicated python module and
+all of them receive and return an Iris cube, working sequentially on the data
+with no interactions between them.  The order
+in which the preprocessor operations is applied is set by default in order to
+minimize the loss of information due to, for example, temporal and spatial
+subsetting or multi-model averaging. Nevertheless, the user is free to change
+such order to address specific scientific requirements, but keeping in mind
+that some operations must be necessarily performed in a specific order. This is
+the case, for instance, for multi-model statistics, which required the model to
+be on a common grid and therefore has to be called after the regridding module.
+
+In this section, each of the preprocessor modules is described in detail
+following the default order in which they are applied:
+
+* Variable derivation
+* CMOR check and dataset-specific fixes
+* Vertical interpolation
+* Land/Sea/Ice masking
+* Horizontal regridding
+* Missing value masking
+* Multi-model statistics
+* Temporal and spatial operations
+* Unit conversion
 
-Features of the ESMValTool Climate data pre-processor are:
-
-* Regridding
-* Geographical area selection
-* Aggregation of data
-* Provenance tracking of the calculations
-* Model statistics
-* Multi-model mean
-* and many more
 
 Variable derivation
 ===================
-Documentation of _derive.py
 
+The _derive.py module allows to derive variables which are not in the CMIP
+standard data request using standard variables as input. The typical use case
+of this operation is the evaluation of a variable which is only available in an
+observational dataset but not in the models. In this case a derivation function
+can be defined by the user in order to calculate the variable and perform the
+comparison. For example, several observational datasets deliver total column
+ozone as observed variable (toz), but CMIP models only provide the ozone 3D
+field. In this case, a derivation function is provided to vertically
+integrate the ozone and obtain total column ozone for direct comparison with
+the observations.
 
-Time manipulation
-=================
-The _time.py module contains the following preprocessor functions:
+The user contributing a new derived variable also needs to define
+a name for it and to provide the corresponding CMOR table. This is to guarantee
+the proper metadata definition is attached to the derived data. Such custom
+CMOR tables are collected as part of the ESMValTool core package. By default, 
```

Comment:
Maybe it would be nice to make "ESMValTool core package" a link to the core github repo?

Reply:
Added.